src/libraries/System.Linq/src/System/Linq/ElementAt.cs

```diff
@@ -63,33 +84,101 @@ public static TSource ElementAt<TSource>(this IEnumerable<TSource> source, int i
                 return partition.TryGetElementAt(index, out bool _);
             }
 
+            if (source is IList<TSource> list)
```

Comment:
Note that this method doesn't test for `IPartition<T>`, unlike `ElementAt(int)`. While this predates your changes, it's likely an omission that we should fix, `ElementAt` and `ElementAtOrDefault` should have equivalent behaviour.

Reply:
> Note that this method doesn't test for `IPartition<T>`

All 4 methods (ElementAt(int), ElementAt(Index), ElementAtOrDefault(int), ElementAtOrDefault(Index)) detect `IPartition<T>`. https://github.com/Dixin/runtime/blob/linq-index-range/src/libraries/System.Linq/src/System/Linq/ElementAt.cs 

Am I missing something?

Reply:
D'oh, the test is literally above the comment, I'm not sure how I didn't see it. Sorry about that.